narwhals/_dask/dataframe.py

```diff
@@ -106,8 +109,21 @@ def select(
             )
             return self._from_native_dataframe(df)
 
-        df = self._native_dataframe.assign(**new_series).loc[:, list(new_series.keys())]
-        return self._from_native_dataframe(df)
+        pd = get_pandas()
+        de = get_dask_expr()
+
+        col_order = list(new_series.keys())
+
+        left_most_series = next(  # pragma: no cover
```

Comment:
this is guaranteed to not end up in `StopIteration` error as if everything was a scalar the previous block would have been entered and returned